app/models/post.rb

```diff
@@ -1,4 +1,21 @@
 class Post < ApplicationRecord
   belongs_to :user
   has_many :comments
+
+  # Calculate the score of the post for the purpose of selecting trending posts
+  # formula is log_10(x) + y * (submission_time - epoch)/45000
+  # where x = max{number of likes + number of comments, 1}, epoch = 1970/1/1,
+  # time difference is in seconds, y = 1 if there is at least one like or
+  # comment, and 0 otherwise
+  def compute_trending_score
+    # TODO(ajyang99) replace 5 with the actual number of likes once like
+    # functionality is implemented
+    num_activities = 5 + self.comments.count
+    x = [num_activities, 1].max
+    y = num_activities > 0 ? 1 : 0
+    submission_time = self.created_at
+    epoch = DateTime.new(1970,1,1)
+    diff = submission_time.to_time - epoch.to_time # in seconds
```

Comment:
I think you can get `diff` by `self.created_at.to_i`. See: https://stackoverflow.com/questions/4492557/convert-ruby-date-to-integer

Reply:
The int converted is in hours and we still have to convert it to seconds. Is this preferable?

Reply:
```
[5] pry(#<PostTest>)> post
=> #<Post:0x00007fd62c45d2b0
 id: 3,
 title: nil,
 content: nil,
 created_at: Sun, 05 Nov 2017 21:07:27 UTC +00:00,
 updated_at: Sun, 05 Nov 2017 21:07:27 UTC +00:00,
 user_id: 4>
[6] pry(#<PostTest>)> post.created_at.to_i
=> 1509916047
```
I think you get seconds, not hours, by `.to_i` on a `DateTime`.

Reply:
`epoch.to_time` would essentially be 0 since it's 0 seconds since the beginning of epoch time.

Reply:
Sorry I misread what you wrote before - uhm would setting epoch to 1970/1/1 be more preferable since "since beginning of time" yields a huge number?